src/normals.c

```diff
@@ -141,14 +144,15 @@ stl_fix_normal_directions(stl_file *stl) {
           (stl, stl->neighbors_start[facet_num].neighbor[j]);
         }
       }
+      const size_t neighbor_index = stl->neighbors_start[facet_num].neighbor[j];
       /* If this edge of the facet is connected: */
-      if(stl->neighbors_start[facet_num].neighbor[j] != -1) {
+      if(neighbor_index != -1 && neighbor_index < number_of_alloc_bytes) {
```

Comment:
Is this the only actual difference that is not "refactoring"? Could you please make the diff simple as that? It makes backport and everything else easier.

Reply:
Please, check again. Thanks